pepper-apis/src/main/java/org/broadinstitute/ddp/json/auth0/Auth0LogEventNode.java

```diff
@@ -0,0 +1,134 @@
+package org.broadinstitute.ddp.json.auth0;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.NodeType.DATETIME;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.NodeType.E_MAIL;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.NodeType.JSON_NODE;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.NodeType.STR;
+
+import java.time.Instant;
+import java.util.Map;
+
+
+import com.google.gson.JsonElement;
+
+
+/**
+ * Contains Auth0 Log Event JSON nodes description which needs to read
+ * from Auth0 Log event Payload (JSON doc).
+ * In this enum defined some logic of JSON nodes parsing (nodes extracted using
+ * class {@link org.broadinstitute.ddp.util.GsonRecursiveReader}.
+ * This approach is used (instead of deserializing JSON to a POJO class) because of the following reasons:
+ * <ul>
+ *     <li>format of Auth0 log event is not same for different types: node "details" differs for
+ *     some types;</li>
+ *     <li>some field which to be defined on top level of node "data" can be defined deeper inside node "details" instead
+ *     (therefore it needs recursively find node by name (for example node "user_id");</li>
+ *     <li>different nodes can hold email value (it can be "email" or "user_name": if "email" not found - read it from "user_name");</li>
+ * </ul>
+ */
+public enum Auth0LogEventNode {
+
+    LOGS("logs", JSON_NODE),
+    LOG_ID("log_id", STR),
+    DATE("date", DATETIME),
+    TYPE("type", STR),
+    CLIENT_ID("client_id", STR),
+    CONNECTION_ID("connection_id", STR),
+    USER_ID("user_id", STR),
```

Comment:
Yes, in this doc it will get "user_id" with prefix "auth0|..".
May be it's ok. I checked other possible log records. For example this one:
`    {
      "log_id": "90020210111084426740006556785942888001718262650720223298",
      "data": {
        "date": "2021-01-11T08:44:24.064Z",
        "type": "sertft",
        "description": "Successful Refresh Token exchange",
        "connection": "Username-Password-Authentication",
        "connection_id": "con_7q6tTpAKRg1fvS5j",
        "client_id": "64YEu4K0XkuCPft4dx9EgdIm9ChxPe2C",
        "client_name": "evgenii-local",
        "ip": "83.102.216.27",
        "user_agent": "Apache-HttpClient/4.5.1 (Java/11.0.9)",
        "details": {
          "tokenCounter": 1,
          "familyId": "nv-EsDeqQhM89i4acu9xiQ"
        },
        "user_id": "auth0|5ffc0fd73ca027006f4fd97a",
        "user_name": "pavel_20210111_10001@broadinstitute.org",
        "audience": "https://ddp-dev.auth0.com/userinfo",
        "scope": "openid profile",
        "log_id": "90020210111084426740006556785942888001718262650720223298"
      }
    },
    `
Here it's only one occurence of "user_id" and it is with prefix.
Seems it always gets "user_id" with prefix and as you said earlier that better to include it in the user_id then user_id data will be consistent. Of course it will be consistent if we always reject prefix (as in 1st implementation). Seems both are ok.